src/engine/loopingcontrol.h

```diff
@@ -32,14 +32,14 @@ class LoopingControl : public EngineControl {
     // process() updates the internal state of the LoopingControl to reflect the
     // correct current sample. If a loop should be taken LoopingControl returns
     // the sample that should be seeked to. Otherwise it returns currentSample.
-    double process(const double dRate,
+    virtual double process(const double dRate,
```

Comment:
If you insist to use a LoopingControlInterface class, I can add one.

But I am not convinced. In this case we test the ReadAheadManager so there are no asuption-cycles. For me the Mixxx code it the part should be clean and easy to read, some ugly hacks in the test are OK for me.  We just need the Stub to pipe in trigger points, It is perfectly fine to do this by virtual override original Loopingcontrol functions. So there is nothing ugly here.  

From the unit test point of view, a stub implements an interface because the original class is not ready or you need to decouple the test from the rest of the code. In this case we are in the lucky situation the original class is ready and already decoupled.